### PR TITLE
Convert List* API calls to Search* API calls for MLflow 2.0 compatibility

### PR DESCRIPTION
**Pull request by dbczumar**
Convert List* API calls to Search* API calls for MLflow 2.0 compatibility. MLflow 2.0 removes ListExperiments and ListRegisteredModels REST & Python APIs, as well as the list_run_infos REST API. This PR accommodates the change by replacing instances of these List API calls with Search API calls.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
     packages = find_packages(),
     zip_safe = False,
     install_requires = [
-          "mlflow>=1.26.0",
+          "mlflow>=1.28.0",
```
Comment (dbczumar) [inline setup.py:20]: SearchExperiments only exists in 1.28.0 and above.

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Review by amesar — DISMISSED** [review]
LGTM

**Review by amesar — APPROVED** [review]
LGTM

**Review by amesar — APPROVED** [review]

**Review by amesar — APPROVED** [review]
LGTM

**Review by amesar — APPROVED** [review]
LGTM

**Pull request merged**